### DISCUSSION TITLE
Setting a filter vs no filter

### DISCUSSION THREAD

**Discussion by Lawkss** [fail2ban/fail2ban#3540] [Q&A]
I dont quite understand filters. 

For example, sshd and postfix-sasl have no filters defined on my install. Yet the jails work and are banning away. 

dovecot and proftpd jails do nothing despite being enabled. Do I need to define the filters for these? 

if no filter is defined, does it use the default filter for that jail? so sshd jail uses sshd filter automatically if not defined? 

I dont understand why roundcube logins are not banned either on any browser the ip shows as "rip=::1, lip=::1," in the logs I think it has trouble detecting the ip? Why does it detect a remote login via browser to roundcube as localhost.

**Comment by sebres** [comment]
> Do I need to define the filters for these?

No. It is historical (older versions of fail2ban did work with filter only). One can specify everything in jail directly.
Although using filter file it may be a bit more safe, especially some complex filters with include etc. Because if it'd interpolate some internal subst-variables, it'll be safe against confusion with subst-variables with same names specified in jail configuration.

> if no filter is defined, does it use the default filter for that jail? 

Yes. Related to jail.conf:
https://github.com/fail2ban/fail2ban/blob/3e6a60f9407596586fe39fa54fd63818c0972b7b/config/jail.conf#L167

One need to set filter to empty value to avoid filter load:
```ini
[some-jail]
filter = 
failregex = ...
...
```

> I dont understand why roundcube logins are not banned either on any browser the ip shows as "rip=::1, lip=::1," in the logs I think it has trouble detecting the ip? 

This is fail2ban repository, so basically the wrong place for such questions.
My assumption - you may have looked in the wrong log file...
For example to look in log file of mail service may be incorrect, since in the chain:
`CLIENT (browser, remote IP) -> WEB-SERVER (roundcube app, local IP) -> MAIL-SVC (local IP)`, the IP that `MAIL-SVC` will notice in log would be local IP, because from its point of view only the last connection matters.
You'd need to look at logs of `WEB-SERVER` or wherever roundcube would log it, not in the mail logs.
